src/base/dof_map_constraints.C

```diff
@@ -431,6 +431,9 @@ private:
         std::vector<bool> is_boundary_node(elem->n_nodes(), false),
           is_boundary_edge(elem->n_edges(), false),
           is_boundary_side(elem->n_sides(), false);
+
+        // We also maintain a separate list of nodeset-based boundary nodes
+        std::vector<bool> is_boundary_nodeset(elem->n_nodes(), false);
```

Comment:
Thanks, didn't know about the distinction on this issue between the different standards.  I wonder if it means that operator[] access is slower for `vector<bool>` in C++11 because they have to do some locking to enforce 23.2.2.